src/app/shared/carousel/carousel.spec.ts

```diff
@@ -0,0 +1,103 @@
+import {DocsAppTestingModule} from '../../testing/testing-module';
+import {Component, ViewChild} from '@angular/core';
+import {CarouselModule} from './carousel-module';
+
+
+describe('HorizontalCarousel', () => {
+  let fixture: ComponentFixture<CarouselTestComponent>;
+  let component: Carousel;
+
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule(
+      {
+        imports: [CarouselModule, DocsAppTestingModule],
+        declarations: [CarouselTestComponent],
+      }
+    ).compileComponents();
+  }));
+
+  beforeEach(fakeAsync(() => {
+    fixture = TestBed.createComponent(CarouselTestComponent);
+    fixture.nativeElement.style.width = '1300px';
+    fixture.detectChanges();
+    flush();
+    component = fixture.componentInstance.carousel;
+  }));
+
+  it('should not show prev nav arrow when instantiated', () => {
+    const navPrevious = fixture.nativeElement.querySelector('.docs-carousel-nav-prev');
+    expect(navPrevious).toBeNull();
+
+    const navNext = fixture.nativeElement.querySelector('.docs-carousel-nav-next');
+    expect(navNext).toBeDefined();
+  });
+
+  it('should show prev nav arrow after increasing index', () => {
+    component.next();
+    fixture.detectChanges();
+
+    expect(component.index).toEqual(1);
+
+    const navPrevious = fixture.nativeElement.querySelector('.docs-carousel-nav-prev');
+    expect(navPrevious).toBeDefined();
+  });
+
+  it('should hide next nav arrow after reaching end of items', () => {
+    expect(component.visibleCards).toBe(4);
+
+    component.next();
+    component.next();
+    component.next();
+    component.next();
+    fixture.detectChanges();
+
+    expect(component.index).toEqual(4);
+
+    const navPrevious = fixture.nativeElement.querySelector('.docs-carousel-nav-next');
+    expect(navPrevious).toBeNull();
+  });
+
+  it('should resize carousel when not all content can be displayed', () => {
+    const carouselWrapper = fixture.nativeElement.querySelector('.docs-carousel-content-wrapper');
+    fixture.nativeElement.style.width = '1350px';
+    window.dispatchEvent(new Event('resize'));
+
+    fixture.detectChanges();
+
+    expect(carouselWrapper.clientWidth).toEqual(1250);
+    expect(component.visibleCards).toEqual(5);
+  });
+
+  it('should not resize carousel when all content can be displayed', () => {
+    fixture.componentInstance.numberOfItems = 2;
+    fixture.detectChanges();
+
+    const carouselWrapper = fixture.nativeElement.querySelector('.docs-carousel-content-wrapper');
+    fixture.nativeElement.style.width = '1350px';
+    window.dispatchEvent(new Event('resize'));
+
+    fixture.detectChanges();
+
+    expect(carouselWrapper.clientWidth).toEqual(500);
+    expect(component.visibleCards).toEqual(2);
+  });
+});
+
+@Component({
+  selector: 'test-carousel',
+  template:
+      `
+    <app-carousel itemWidth="250">
+      <div carousel-item class="docs-carousel-item-container"
+           *ngFor="let i of [].constructor(numberOfItems) "></div>
```

Comment:
Hmmm, this might require a sizeable refactor so maybe add a follow up to investigate:
The content slider accessibility guide's [checklist](https://inclusive-components.design/a-content-slider/#checklist) suggested using " list markup to group the slides together. Then screen reader users in 'browse' mode can use list navigation shortcuts to traverse them."

I think you can inspect the styles on the ul/li elements in this [demo](https://codepen.io/heydon/full/xPWOLp) or what's [listed in the guide](https://inclusive-components.design/a-content-slider/#slides) to get it working.

```
ul {
  display: flex;
}

li {
  list-style: none;
  flex: 0 0 100%;
}
```

Reply:
I tried implementing this but it got a bit messy so I think I will do this in a followup PR along with the cdkObserveContent